include/fastrtps/participant/Participant.h

```diff
@@ -86,6 +87,12 @@ class RTPS_DllAPI Participant
     bool get_remote_writer_info(const rtps::GUID_t& writerGuid, rtps::WriterProxyData& returnedInfo);
 
     bool get_remote_reader_info(const rtps::GUID_t& readerGuid, rtps::ReaderProxyData& returnedInfo);
+
+    /**
+     * Get the resource event for this participant
+     */
+    rtps::ResourceEvent& get_resource_event() const;
```

Comment:
Why did you need this? Where is it used? I think this function shouldn't be in the public API.

Reply:
I needed it to be able to initialize the deadline timer in ```PublisherImpl``` (same for ```SubscriberImpl```):

https://github.com/eProsima/Fast-RTPS/blob/ac9b62aad764ccccf41613bb37aa4569975305ab/src/cpp/publisher/PublisherImpl.cpp#L66-L69

Reply:
But `PublisherImpl` contains a pointer to `ParticipantImpl`, not to `Participant`:

https://github.com/eProsima/Fast-RTPS/blob/feature/DeadlineQos/src/cpp/publisher/PublisherImpl.h#L141

Reply:
Ah OK, I see now that this is not necessary. I think I was trying different ways to get the resource event and didn't remove this. I've done it now.